### ISSUE TITLE
Embedding in a `UITableViewCell` prevents `didEndDisplayingCell` to be called.

### ISSUE THREAD

**Issue by MightyFine** [gilesvangruisen/Swift-YouTube-Player#154]
Embedding the player in a `UITableViewCell`, when the user interacts with it to start a video, the table view's delegate no longer received `didEndDisplayingCell` calls for the cell with the player in it.

Some notes on investigating this:

- The issue does not occur when playback is started programmatically using the `play()` function.
- The same issue occurs in the following framework which uses a very similar implementation in Objective C https://github.com/youtube/youtube-ios-player-helper and there are two threads in their issues where people have reported this with no answers.
- The issue appears to resolve if we push or present from the housing view controller and navigate back. The video in this case is paused by the system (?) and can be interacted with without this happening again.
- This issue also occurs when loading URLs other than the embedded YouTube link. Loading https://www.google.com, when interacting with some elements on the page, this doesn't occur, but when interacting with others (such as the top right services menu), it does.
- When swapping `UIWebView` out for `WKWebView`, the problem persists.

In conclusion, this appears to be a deeper underlying issue with how web views in iOS behave when loading various types of content.

I haven't managed to figure it out, so if anyone has any ideas, that'd be both helpful, and interesting.

**Comment by ranamagdy87** [comment]
Same Problem
